subsystems/sidecar/index.js

```diff
@@ -527,8 +541,19 @@ class Sidecar extends ReadyResource {
       return
     }
 
+    if (!hard) {
+      const clients = this.uniqueClients.filter(c => c?.userData?.state?.options?.type === 'terminal')
+      if (this.verbose) console.log(`Soft-restarting ${clients.length} terminal app(s)`)
+      for (const client of clients) client.userData.message({ type: 'pear/restart' })
+
+      // Wait for 'pear/restart' messages to send and get handled by clients
+      // TODO: Figure out how to cleanly flush this buffer
+      await new Promise(resolve => setTimeout(resolve, 1500))
```

Comment:
don't understand why this is needed, but if it is you'll need an ipc call from client a counter - but ideally don't do this at all, just wait for sidecar to shutdown (can't be any clients if it's shut) and then trigger restarts